### PR TITLE
593 file cannot be uploaded at the dialogue window task

### PR DESCRIPTION
**Pull request by 0ffffh**

### CODE REVIEW

File: goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultFilesStorageService.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+    @Autowired
+    public DefaultFilesStorageService(FilesStorageProperties fileStorageProperties, AttachmentRepository attachmentRepository) {
+        this.fileStorageLocation = Paths.get(fileStorageProperties.getUploadStorage())
+                .toAbsolutePath().normalize();
+        this.attachmentRepository = attachmentRepository;
+        try {
+            Files.createDirectories(this.fileStorageLocation);
+        } catch (Exception e) {
+            throw new RuntimeException("Could not create the directory where the uploaded files will be stored.", e);
```
Comment (TarasOrlovskyi) [inline goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultFilesStorageService.java:47]: you should have thrown a custom exception

Comment (0ffffh) [reply goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultFilesStorageService.java]: ok

File: goods-partner-backend/src/main/java/com/goodspartner/entity/User.java [hunk 2]
```diff
@@ -50,8 +50,14 @@ public class User {
...
-        DRIVER,
-        LOGISTICIAN
+
+        ADMIN("ROLE_ADMIN"),
+        DRIVER("ROLE_DRIVER"),
+        LOGISTICIAN("ROLE_LOGISTICIAN");
+        private final String name;
+
+        UserRole(String name) {
+            this.name = name;
```
[the remark this answers is not in the document]
Comment (0ffffh) [reply goods-partner-backend/src/main/java/com/goodspartner/entity/User.java]: fixed

File: goods-partner-backend/src/main/java/com/goodspartner/configuration/converter/StringToTaskConverter.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+@RequiredArgsConstructor
+public class StringToTaskConverter implements Converter<String, TaskDto> {
+    private final ObjectMapper objectMapper;
+
+    @Override
+    public TaskDto convert(String source) {
+        try {
+            return objectMapper.readValue(source, TaskDto.class);
+        } catch (IOException e) {
+            throw new RuntimeException(e.getMessage());
```
[the remark this answers is not in the document]
Comment (0ffffh) [reply goods-partner-backend/src/main/java/com/goodspartner/configuration/converter/StringToTaskConverter.java]: ok

File: goods-partner-backend/src/main/java/com/goodspartner/entity/Attachment.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    @Column(name = "id", updatable = false)
+    @GeneratedValue(generator = "UUID")
+    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
+    private UUID id;
+
+    @Column(name = "file_name_original")
+    private String fileNameOriginal;
+
+    @Column(name = "file_type")
+    private String fileType;
```
[the remark this answers is not in the document]
Comment (0ffffh) [reply goods-partner-backend/src/main/java/com/goodspartner/entity/Attachment.java:40]: I beg to differ

**Pull request closed without merging**